target_snowflake/flattening.py

```diff
@@ -0,0 +1,115 @@
+import collections
+import inflection
+import itertools
+import json
+import re
+
+
+def flatten_key(k, parent_key, sep):
+    """
+
+    Params:
+        k:
+        parent_key:
+        sep:
+
+    Returns:
+    """
+    full_key = parent_key + [k]
+    inflected_key = full_key.copy()
+    reducer_index = 0
+    while len(sep.join(inflected_key)) >= 255 and reducer_index < len(inflected_key):
+        reduced_key = re.sub(r'[a-z]', '', inflection.camelize(inflected_key[reducer_index]))
+        inflected_key[reducer_index] = \
+            (reduced_key if len(reduced_key) > 1 else inflected_key[reducer_index][0:3]).lower()
+        reducer_index += 1
+
+    return sep.join(inflected_key)
+
+
+# pylint: disable=invalid-name,dangerous-default-value
+def flatten_schema(d, parent_key=[], sep='__', level=0, max_level=0):
```

Comment:
Can we address the disabled pylint complaints above the function name?
 For `dangerous-default-value` we can get rid of it by changing the value of `parent_key` to be like this

```python
def flatten_schema(d, parent_key=None, sep='__', level=0, max_level=0)
    if parent_key is None:
           parent_key = []
```

Reply:
fixed by 786ed43a49cb2754c56b74ab50d0ee5b3d5421e2

Reply:
fixed by 786ed43a49cb2754c56b74ab50d0ee5b3d5421e2